File: app/src/main/java/com/antitheft/pro/SettingsActivity.java
package com.antitheft.pro;

import android.app.Activity;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.CheckBox;
import android.widget.EditText;
import android.widget.LinearLayout;
import android.widget.TextView;
import android.widget.Toast;

public class SettingsActivity extends Activity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        
        try {
            // إنشاء layout برمجياً
            createSettingsLayoutProgrammatically();
            
            Toast.makeText(this, "صفحة الإعدادات جاهزة", Toast.LENGTH_SHORT).show();
            
        } catch (Exception e) {
            Toast.makeText(this, "خطأ: " + e.getMessage(), Toast.LENGTH_LONG).show();
            e.printStackTrace();
            finish();
        }
    }
    
    private void createSettingsLayoutProgrammatically() {
        try {
            // إنشاء LinearLayout رئيسي
            LinearLayout mainLayout = new LinearLayout(this);
            mainLayout.setOrientation(LinearLayout.VERTICAL);
            mainLayout.setPadding(50, 50, 50, 50);
            
            // عنوان
            TextView titleText = new TextView(this);
            titleText.setText("الإعدادات");
            titleText.setTextSize(24);
            titleText.setPadding(0, 0, 0, 30);
            mainLayout.addView(titleText);
            
            // إعدادات الأمان
            TextView securityTitle = new TextView(this);
            securityTitle.setText("إعدادات الأمان");
            securityTitle.setTextSize(18);
            securityTitle.setPadding(0, 0, 0, 20);
            mainLayout.addView(securityTitle);
            
            // خيارات الأمان
            final CheckBox pinLockBox = new CheckBox(this);
            pinLockBox.setText("تفعيل قفل الرقم السري");
            mainLayout.addView(pinLockBox);
            
            final CheckBox intruderBox = new CheckBox(this);
            intruderBox.setText("تفعيل كشف المتطفلين");
            mainLayout.addView(intruderBox);
            
            final CheckBox locationBox = new CheckBox(this);
            locationBox.setText("تفعيل تتبع الموقع");
            mainLayout.addView(locationBox);
            
            // بريد الإشعارات
            TextView emailTitle = new TextView(this);
            emailTitle.setText("بريد الإشعارات");
            emailTitle.setTextSize(18);
            emailTitle.setPadding(0, 30, 0, 10);
            mainLayout.addView(emailTitle);
            
            final EditText emailField = new EditText(this);
            emailField.setHint("أدخل بريدك الإلكتروني");
            emailField.setPadding(20, 20, 20, 20);
            mainLayout.addView(emailField);
            
            // زر الحفظ
            Button saveButton = new Button(this);
            saveButton.setText("حفظ الإعدادات");
            saveButton.setPadding(20, 20, 20, 20);
            saveButton.setOnClickListener(new View.OnClickListener() {
                @Override
                public void onClick(View v) {
                    saveSettings(pinLockBox, intruderBox, locationBox, emailField);
                }
            });
            mainLayout.addView(saveButton);
            
            // زر العودة
            Button backButton = new Button(this);
            backButton.setText("عودة");
            backButton.setPadding(20, 20, 20, 20);
            backButton.setOnClickListener(new View.OnClickListener() {
                @Override
                public void onClick(View v) {
                    finish();
                }
            });
            mainLayout.addView(backButton);
            
            // تعيين Layout
            setContentView(mainLayout);
            
        } catch (Exception e) {
            Toast.makeText(this, "خطأ في إنشاء واجهة الإعدادات: " + e.getMessage(), Toast.LENGTH_SHORT).show();
            e.printStackTrace();
        }
    }
    
    private void saveSettings(CheckBox pinLockBox, CheckBox intruderBox, CheckBox locationBox, EditText emailField) {
        try {
            // محاكاة حفظ الإعدادات
            String message = "تم حفظ الإعدادات:\n";
            
            if (pinLockBox.isChecked()) {
                message += "- قفل الرقم السري مفعل\n";
            }
            
            if (intruderBox.isChecked()) {
                message += "- كشف المتطفلين مفعل\n";
            }
            
            if (locationBox.isChecked()) {
                message += "- تتبع الموقع مفعل\n";
            }
            
            String email = emailField.getText().toString().trim();
            if (!email.isEmpty()) {
                message += "- بريد الإشعارات: " + email;
            }
            
            Toast.makeText(this, message, Toast.LENGTH_LONG).show();
            
        } catch (Exception e) {
            Toast.makeText(this, "خطأ في حفظ الإعدادات: " + e.getMessage(), Toast.LENGTH_SHORT).show();
            e.printStackTrace();
        }
    }
}

File: .expo/app/src/main/java/com/antitheft/pro/activities/SettingsActivity.kt
package com.antitheft.pro.activities

import android.content.Intent
import android.os.Bundle
import androidx.appcompat.app.AlertDialog
import androidx.appcompat.app.AppCompatActivity
import com.antitheft.pro.R
import com.antitheft.pro.databinding.ActivitySettingsBinding
import com.antitheft.pro.utils.LanguageManager
import com.antitheft.pro.utils.PreferenceManager

class SettingsActivity : AppCompatActivity() {
    
    private lateinit var binding: ActivitySettingsBinding
    private lateinit var prefManager: PreferenceManager
    private lateinit var languageManager: LanguageManager

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivitySettingsBinding.inflate(layoutInflater)
        setContentView(binding.root)
        
        prefManager = PreferenceManager(this)
        languageManager = LanguageManager(this)
        
        setupUI()
    }
    
    private fun setupUI() {
        // عنوان الصفحة
        supportActionBar?.title = getString(R.string.settings)
        supportActionBar?.setDisplayHomeAsUpEnabled(true)
        
        // تحميل الإعدادات الحالية
        binding.apply {
            switchPinLock.isChecked = prefManager.isPinEnabled()
            switchIntruderDetection.isChecked = prefManager.isIntruderDetectionEnabled()
            switchLocationTracking.isChecked = prefManager.isLocationTrackingEnabled()
            switchWifiScanner.isChecked = prefManager.isWifiScannerEnabled()
            
            val currentEmail = prefManager.getNotificationEmail()
            editTextNotificationEmail.setText(currentEmail)
            
            val currentLanguage = languageManager.getCurrentLanguage()
            textViewCurrentLanguage.text = getLanguageName(currentLanguage)
            
            // PIN Lock
            switchPinLock.setOnCheckedChangeListener { _, isChecked ->
                if (isChecked && prefManager.getPin().isEmpty()) {
                    // إنشاء PIN جديد
                    startActivity(Intent(this@SettingsActivity, SetPinActivity::class.java))
                } else {
                    prefManager.setPinEnabled(isChecked)
                }
            }
            
            buttonChangePin.setOnClickListener {
                if (prefManager.isPinEnabled()) {
                    startActivity(Intent(this@SettingsActivity, SetPinActivity::class.java))
                }
            }
            
            // Intruder Detection
            switchIntruderDetection.setOnCheckedChangeListener { _, isChecked ->
                prefManager.setIntruderDetectionEnabled(isChecked)
            }
            
            // Location Tracking
            switchLocationTracking.setOnCheckedChangeListener { _, isChecked ->
                prefManager.setLocationTrackingEnabled(isChecked)
            }
            
            // WiFi Scanner
            switchWifiScanner.setOnCheckedChangeListener { _, isChecked ->
                prefManager.setWifiScannerEnabled(isChecked)
            }
            
            // حفظ بريد الإشعارات
            buttonSaveEmail.setOnClickListener {
                val email = editTextNotificationEmail.text.toString().trim()
                prefManager.setNotificationEmail(email)
                android.widget.Toast.makeText(
                    this@SettingsActivity,
                    R.string.success,
                    android.widget.Toast.LENGTH_SHORT
                ).show()
            }
            
            // تغيير اللغة
            buttonChangeLanguage.setOnClickListener {
                showLanguageDialog()
            }
        }
    }
    
    private fun showLanguageDialog() {
        val languages = arrayOf(
            "العربية" to "ar",
            "English" to "en",
            "Ελληνικά" to "el",
            "Deutsch" to "de",
            "Français" to "fr",
            "Español" to "es",
            "Italiano" to "it",
            "Português" to "pt",
            "Nederlands" to "nl",
            "Polski" to "pl",
            "Română" to "ro",
            "Čeština" to "cs",
            "Svenska" to "sv",
            "Magyar" to "hu",
            "Dansk" to "da",
            "Suomi" to "fi",
            "Slovenčina" to "sk",
            "Български" to "bg",
            "Hrvatski" to "hr",
            "Slovenščina" to "sl",
            "Lietuvių" to "lt",
            "Latviešu" to "lv",
            "Eesti" to "et",
            "Malti" to "mt",
            "Gaeilge" to "ga"
        )
        
        val languageNames = languages.map { it.first }.toTypedArray()
        val currentLanguage = languageManager.getCurrentLanguage()
        val currentIndex = languages.indexOfFirst { it.second == currentLanguage }
        
        AlertDialog.Builder(this)
            .setTitle(R.string.language)
            .setSingleChoiceItems(languageNames, currentIndex) { dialog, which ->
                val selectedLanguage = languages[which].second
                languageManager.setLanguage(selectedLanguage)
                
                // إعادة تشغيل Activity
                recreate()
                dialog.dismiss()
            }
            .setNegativeButton(R.string.cancel, null)
            .show()
    }
    
    private fun getLanguageName(languageCode: String): String {
        return when (languageCode) {
            "ar" -> "العربية"
            "en" -> "English"
            "el" -> "Ελληνικά"
            "de" -> "Deutsch"
            "fr" -> "Français"
            "es" -> "Español"
            "it" -> "Italiano"
            "pt" -> "Português"
            "nl" -> "Nederlands"
            "pl" -> "Polski"
            "ro" -> "Română"
            "cs" -> "Čeština"
            "sv" -> "Svenska"
            "hu" -> "Magyar"
            "da" -> "Dansk"
            "fi" -> "Suomi"
            "sk" -> "Slovenčina"
            "bg" -> "Български"
            "hr" -> "Hrvatski"
            "sl" -> "Slovenščina"
            "lt" -> "Lietuvių"
            "lv" -> "Latviešu"
            "et" -> "Eesti"
            "mt" -> "Malti"
            "ga" -> "Gaeilge"
            else -> "English"
        }
    }
    
    override fun onSupportNavigateUp(): Boolean {
        finish()
        return true
    }
}
